go/cmd/dolt/commands/diff.go

```diff
@@ -613,7 +619,41 @@ func diffRows(
 
 	defer rowIter.Close(sqlCtx)
 
-	err = writeDiffResults(sqlCtx, sch, unionSch, rowIter, rowWriter)
+	if dArgs.skinny {
+		modifiedColNames, err := getModifiedCols(sqlCtx, rowIter, unionSch, sch)
+		if err != nil {
+			return errhand.BuildDError("Error running diff query:\n%s", query).AddCause(err).Build()
+		}
+
+		// instantiate a new schema that only contains the columns with changes
+		var filteredUnionSch sql.Schema
+		for _, s := range unionSch {
+			for colName := range modifiedColNames {
+				if strings.Contains(s.Name, colName) {
+					filteredUnionSch = append(filteredUnionSch, s)
+				}
+			}
+		}
+
+		// instantiate a new RowWriter with the new schema that only contains the columns with changes
+		rowWriter, err = dw.RowWriter(ctx, td, filteredUnionSch)
+		if err != nil {
+			return errhand.VerboseErrorFromError(err)
+		}
+		defer rowWriter.Close(ctx)
+
+		// reset the row iterator
+		rowIter.Close(sqlCtx)
+		_, rowIter, err = se.Query(sqlCtx, query)
+		if sql.ErrSyntaxError.Is(err) {
+			return errhand.BuildDError("Failed to parse diff query. Invalid where clause?\nDiff query: %s", query).AddCause(err).Build()
+		} else if err != nil {
+			return errhand.BuildDError("Error running diff query:\n%s", query).AddCause(err).Build()
+		}
+		defer rowIter.Close(sqlCtx)
+	}
```

Comment:
Not sure if this is the correct way to reset the row iterator. Any feedback would be greatly appreciated!

Reply:
yup, this looks great! 

Would be good to move that `defer` up to line 648 to make sure that `Close` call always gets registered to be run. Currently, it won't run if any of that error code above it returns from the function. Not a huge deal for correctness, but definitely good practice to tidy up.